projects/go-lib/src/lib/components/go-button-group/go-button-group.component.ts

```diff
@@ -0,0 +1,54 @@
+import { AfterContentInit, Component, ContentChildren, Input, OnChanges, QueryList } from '@angular/core';
+import { GoButtonComponent } from '../go-button/go-button.component';
+
+@Component({
+  selector: 'go-button-group',
+  templateUrl: './go-button-group.component.html'
+})
+export class GoButtonGroupComponent implements OnChanges, AfterContentInit {
+
+  @Input() buttonDisabled: boolean = false;
+  @Input() buttonVariant: GoButtonComponent['buttonVariant'] = 'primary';
+
+  @ContentChildren(GoButtonComponent) buttons: QueryList<GoButtonComponent>;
+
+  ngOnChanges(): void {
+    this.processButtons();
+  }
+
+  ngAfterContentInit(): void {
+    this.processButtons();
+  }
+
+  private processButtons(): void {
+    if (this.buttons) {
+      this.buttons.forEach((btn: GoButtonComponent, i: number) => {
+        btn.groupPosition = this.determinePosition(i);
+        btn.buttonDisabled = this.buttonDisabled;
+        btn.buttonVariant = this.buttonVariant;
+        btn.ngOnChanges();
+      });
+    }
+  }
+
+  private determinePosition(i: number): GoButtonComponent['groupPosition'] {
+    let position: GoButtonComponent['groupPosition'] = null;
```

Comment:
If `determinePosition` is expected to always return either `first`, `last`, or `middle`, you could rewrite this function without using the `position` variable and reduce complexity greatly:

```Javascript
if (this.buttons.length > 1) {
  if (i === 0) {
    return 'first';
  }
  
  if (i === this.buttons.length -1) {
    return 'last';
  }
  
  return 'middle';
}
```

Not sure if this is proper formatting for this codebase, but it _might_ remove the cognitive complexity warning.

Reply:
Technically, it can also return `null`. If there is only 1 `go-button` in the `go-button-group` then we will want to return `null` so that it doesn't apply any of the modifier classes.

Reply:
Good catch, I added a `return null` to the end.

Reply:
I updated it to what you put, I do think that's a little bit more concise.